server/utils.go

```diff
@@ -176,3 +248,15 @@ func mapKeys[K comparable, V any](m map[K]V) []K {
 	}
 	return keys
 }
+
+type SimpleClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// NewClient creates a SimpleClient intended for one-off requests, like getPushProxyVersion.
+// If we end up needing something more long term, we should consider storing it.
+func NewClient() *http.Client {
```

Comment:
The reason I mentioned the `init()` function is because this is technically racy as we are modifying a global object. It would probably work fine for our current use case but it may cause problems in the future if anything else relies on `DefaultClient`.

Reply:
When returning a client struct with the timeout set like we are now, I don't think there's still an issue? But let me know and I'll fix it (and I'll learn something).

Reply:
Yep, now it should work fine, just sharing the default transport this way.